presto-main/src/main/java/com/facebook/presto/sql/planner/PartitioningScheme.java

```diff
@@ -29,10 +29,40 @@
 
 public class PartitioningScheme
 {
+    public enum Replication
+    {
+        NONE {
```

Comment:
instead of overriding method in each `Replication` you could use:
```
public enum Replication {
   NONE(true, Integer.MAX_VALUE), REPLICATE_NULLS(false, 1);

   private final boolean noReplication;
   private final int maxReplicationColumns;

   Replication(...) {
   }
   
   public boolean isNoReplication() {...}
   public int isCompatibleWithColumns(...) {...}
   
}
```